simplestore/src/main/java/com/uber/simplestore/impl/SimpleStoreFactory.java

```diff
@@ -63,11 +63,57 @@ public static SimpleStore create(DirectoryProvider directoryProvider, String nam
    */
   public static SimpleStore create(
       DirectoryProvider directoryProvider, String namespace, NamespaceConfig config) {
+    return provideStore(directoryProvider, namespace, config, false);
+  }
+
+  /**
+   * If doesn't exist an open store instance for the namespace, will create and return a store for a
+   * namespace with the default configuration. If exist an open store instance for the namespace,
+   * will return a store for a namespace with the default configuration.
+   *
+   * @param directoryProvider to store the files in
+   * @param namespace forward-slash delimited logical address
+   * @return open store
+   */
+  public static SimpleStore getOrCreate(DirectoryProvider directoryProvider, String namespace) {
+    return provideStore(directoryProvider, namespace, NamespaceConfig.DEFAULT, true);
+  }
+
+  /**
+   * If doesn't exist an open store instance for the namespace, will create and return a store for a
+   * namespace. If exist an open store instance for the namespace, will return a store for a
+   * namespace.
+   *
+   * @param directoryProvider to store the files in
+   * @param namespace forward-slash delimited logical address
+   * @param config to use
+   * @return open store
+   */
+  public static SimpleStore getOrCreate(
+      DirectoryProvider directoryProvider, String namespace, NamespaceConfig config) {
+    return provideStore(directoryProvider, namespace, config, true);
+  }
+
+  /**
+   * Internal method to obtain a store for a namespace.
+   *
+   * @param directoryProvider to store the files in
+   * @param namespace forward-slash delimited logical address
+   * @param config to use
+   * @param allowOpenInstance configure of the create method will throw a IllegalStateException if
+   *     already exists an open instance for the used namespace
```

Comment:
Nit : allowMultipleInstances or something a little bit more descriptive of why we're allowing multiple instances of the namespace

Reply:
We are not allowing multiple instances for the store. 
With this new method, if exists an open instance for the namespace, this will return the opened method and not create a new one.
I'm thinking in `reuseAlreadyOpenedInstances` What do you think?

Reply:
I see, `reuseAlreadyOpenedInstances ` sounds good, lets use that. The other name did give the impression that you wanted to create / use another instance

Reply:
Perfect. I will change the docs to make it clearer